lib/diagnostic.js

```diff
@@ -6,7 +6,13 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
-
+  previous.forEach(function callback(currentValue, index, array) {
+    current = previous[0]
+    for (let i = 1; i < previous.length; i++) {
```

Comment:
This would definitely work on its own, but we don't need to use a `for loop` since this function is going to be passed to a `reduce` function.